core/controllers/feedback.py

```diff
@@ -282,3 +282,29 @@ def post(self):
         email_manager.send_suggestion_email(
             exploration.title, exploration.id, suggestion.author_id,
             exploration_rights.owner_ids)
+
+
+class InstantFeedbackEmailHandler(base.BaseHandler):
+    """Handles task of sending feedback message emails instantsly."""
+
+    def post(self):
+        payload = json.loads(self.request.body)
+        user_id = payload['user_id']
+        reference = payload['reference']
+        message = feedback_services.get_message(
+            reference['exploration_id'], reference['thread_id'],
+            reference['message_id'])
+        exploration = exp_services.get_exploration_by_id(
+            reference['exploration_id'])
+        thread = feedback_services.get_thread(
+            reference['exploration_id'], reference['thread_id'])
+
+        if 'updated_status' in payload:
```

Comment:
So, in that case, you would be sending two emails, one for the status change, and one for the message (if it's non-empty)? If so, I think that's actually better -- I had thought we'd combine these two into a single email, but it makes sense for them to be separate.

That said, the design is indeed a bit garbled. Here's what I think should happen:
- There should be a single function in feedback_services that coordinates which emails get sent to which people (currently I think this is basically `add_message_to_email_buffer`). That should be the single place where all these decisions are made (and the check for "if we need to send a status message, send it" should also happen there).
- Actually, thinking through it a bit more, why do we need this handler anyway? Can't that function just call email_manager.send_instant_feedback_message_email() directly? I can see why you'd need a dedicated URL for deferred tasks, but I don't think this one counts...